server/plugin/api_test.go

```diff
@@ -860,15 +863,57 @@ func Test_handleDateTimeSelection(t *testing.T) {
 			},
 			userID: "mock-userID",
 		},
+		"Error while scheduling the job": {
+			httpTest: httpTestJSON,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("/api/v1%s", PathSetDateTime),
+				Body:   getHandleDateTimeSelectionRequestBody("2022-09-23", "", "Date"),
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+			userID:         "mock-userID",
+			scheduleJobErr: errors.New("error while scheduling the job"),
+		},
+		"Failed to get direct channel": {
+			httpTest: httpTestJSON,
+			request: testutils.Request{
+				Method: http.MethodPost,
+				URL:    fmt.Sprintf("/api/v1%s", PathSetDateTime),
+				Body: model.PostActionIntegrationRequest{
+					Context: map[string]interface{}{
+						"selected_option": "mockOption",
+					},
+				},
+			},
+			expectedResponse: testutils.ExpectedResponse{
+				StatusCode: http.StatusOK,
+			},
+			userID:                "mock-userID",
+			getDirectChannelError: &model.AppError{},
+		},
 	} {
 		t.Run(name, func(t *testing.T) {
+			mockInterval := int64(1000)
 			p := new(Plugin)
 
 			mockAPI := &plugintest.API{}
 			mockAPI.On("GetBundlePath").Return("mockString", nil)
 			mockAPI.On("LogDebug", testutils.GetMockArgumentsWithType("string", 7)...).Return("LogDebug error")
-			mockAPI.On("LogError", testutils.GetMockArgumentsWithType("string", 6)...).Return("LogError error")
+			mockAPI.On("LogError", testutils.GetMockArgumentsWithType("string", 5)...).Return()
 			mockAPI.On("UpdatePost", mock.AnythingOfType("*model.Post")).Return(nil, nil)
+
+			mockAPI.On("GetConfig").Return(&model.Config{
```

Comment:
Instead of doing all this, use monkey patch and mock the return type of `p.scheduleJob`.